### PR TITLE
[flare] Include logs with the existing subdirectory tree

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

If the user creates subdirectories in the agent's log directory,
and for example moves agent logs to these subdirectories, this
change allows the log files to be copied in the flare.

In addition, add the permissions of the log directory itself (ex. `/var/log/datadog/`), which were missing from `permissions.log`.

### Motivation

Previously, in this scenario, log files in these subdirectories
could overwrite the log files in the root log directory
with the same basename.

For example:

```
/var/log/datadog/agent.log
/var/log/datadog/archive/agent.log
```

would previously, in the flare, result in (only one log file included):

```
<unzipped_flare>/logs/agent.log   // file copied from /var/log/datadog/archive/agent.log, /var/log/datadog/agent.log is not included in flare
```

now correctly results in:

```
<unzipped_flare>/logs/agent.log
<unzipped_flare>/logs/archive/agent.log
```

### Describe how to test your changes

Added a unit test.
To QA:
1. create the `archive` subdirectory in the agent's log directory, and create `agent.log` in that directory, as described in the example above
2. generate a flare and inspect it: the `logs` directory should include both `agent.log` and `archive/agent.log`. `permissions.log` should include the permissions to the agent's log dir (e.g. `/var/log/datadog` on Linux)

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.

### CODE REVIEW

File: pkg/flare/archive_test.go [hunk 1]
```diff
@@ -247,6 +249,35 @@ func TestCleanDirectoryName(t *testing.T) {
 	assert.True(t, !directoryNameFilter.MatchString(cleanedHostname))
 }
 
+func TestZipLogFiles(t *testing.T) {
+	srcDir, err := ioutil.TempDir("", "logs")
+	require.NoError(t, err)
+	// defer os.RemoveAll(srcDir)
+	dstDir, err := ioutil.TempDir("", "TestZipLogFiles")
+	require.NoError(t, err)
+	// defer os.RemoveAll(dstDir)
```
Comment (mx-psi) [inline pkg/flare/archive_test.go:258]:
```suggestion

	dstDir, err := ioutil.TempDir("", "TestZipLogFiles")
	require.NoError(t, err)
```

**Review by mx-psi — APPROVED** [review]

File: pkg/flare/archive_test.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/flare/archive_test.go]: Good catch, I commented these out to troubleshoot but these are actually meant to be uncommented.

Fixed it, could you have another look?

File: pkg/flare/archive.go [hunk 2]
```diff
@@ -401,7 +401,8 @@ func zipLogFiles(tempDir, hostname, logFilePath string, permsInfos permissionsIn
 		}
 
 		if filepath.Ext(f.Name()) == ".log" || getFirstSuffix(f.Name()) == ".log" {
-			dst := filepath.Join(tempDir, hostname, "logs", f.Name())
+			baseName := strings.Replace(src, logFileDir, "", 1)
```
Comment (vickenty) [inline pkg/flare/archive.go:404]: `filepath.Rel` might be better?

Comment (mx-psi) [reply pkg/flare/archive.go:404]: ➕ to this

Comment (olivielpeau) [reply pkg/flare/archive.go]: good suggestion, I've addressed this

**Comment by olivielpeau** [comment]
I've addressed comments + made a small additional change to add permission info of the log directory itself (PR description updated), ready for another review @vickenty @mx-psi

**Comment by mx-psi** [comment]
@olivielpeau unit tests are panicking (you can't pass `nil` as the last argument anymore) can you fix them?

**Comment by olivielpeau** [comment]
Sorry, should've checked this before I asked for another review. Should be fixed now.

**Review by mx-psi — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Pull request merged**